BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift

```diff
@@ -106,186 +111,190 @@ public class TransactionConfirmationStore: ObservableObject {
+             let gasValue = BDouble(gasFee.fee),
+             BDouble(gasBalance) > gasValue {
+            state.isBalanceSufficient = true
+          } else {
+            state.isBalanceSufficient = false
+          }
+          
+          state.totalFiat = totalFiat(value: state.value, tokenSymbol: state.symbol, gasValue: state.gasValue, gasSymbol: state.gasSymbol, assetRatios: assetRatios, currencyFormatter: currencyFormatter)
         }
-        updateState()
-        self.isLoading = false
+      case let .erc721Transfer(details):
+        state.symbol = details.fromToken?.symbol ?? ""
+        state.value = details.fromAmount
       }
     }
   }
-
-  private func fetchGasEstimation1559() {
-    ethTxManagerProxy.gasEstimation1559() { [weak self] gasEstimation in
-      self?.gasEstimation1559 = gasEstimation
+  
+  private func totalFiat(
+    value: String,
+    tokenSymbol: String,
+    gasValue: String,
+    gasSymbol: String,
+    assetRatios: [String: Double],
+    currencyFormatter: NumberFormatter
+  ) -> String {
+    if let ratio = assetRatios[tokenSymbol.lowercased()], let gasRatio = assetRatios[gasSymbol.lowercased()] {
+      let amount = (Double(value) ?? 0.0) * ratio
+      let gasAmount = (Double(gasValue) ?? 0.0) * gasRatio
+      let totalFiat = currencyFormatter.string(from: NSNumber(value: amount + gasAmount)) ?? "$0.00"
+      return totalFiat
     }
+    return "$0.00"
   }
 
-  private func fetchTransactions(completion: @escaping (() -> Void)) {
-    walletService.selectedCoin { [weak self] coin in
-      guard let self = self else { return }
-      self.keyringService.keyringInfo(coin.keyringId) { keyring in
-        var pendingTransactions: [BraveWallet.TransactionInfo] = []
-        let group = DispatchGroup()
-        for info in keyring.accountInfos {
-          group.enter()
-          self.txService.allTransactionInfo(info.coin, from: info.address) { tx in
-            defer { group.leave() }
-            pendingTransactions.append(contentsOf: tx.filter { $0.txStatus == .unapproved })
+  @MainActor private func fetchTransactions() async -> [BraveWallet.TransactionInfo] {
+    var allKeyrings: [BraveWallet.KeyringInfo] = []
+    allKeyrings = await withTaskGroup(
+      of: BraveWallet.KeyringInfo.self,
+      returning: [BraveWallet.KeyringInfo].self,
+      body: { group in
+        for coin in WalletConstants.supportedCoinTypes {
+          group.addTask {
+            await self.keyringService.keyringInfo(coin.keyringId)
           }
         }
-        group.notify(queue: .main) {
-          self.transactions = pendingTransactions
-          completion()
+        var allKeyrings: [BraveWallet.KeyringInfo] = []
+        for await keyring in group {
+          allKeyrings.append(keyring)
         }
+        return allKeyrings
       }
-    }
+    )
+    
+    var pendingTransactions: [BraveWallet.TransactionInfo] = []
+    pendingTransactions = await withTaskGroup(
```

Comment:
Similar here about this being re-used in a couple spots now, maybe makes sense as a helper function in an extension on `BraveWalletTxService`